docs/rfc/rfc-3.adoc

```diff
@@ -0,0 +1,119 @@
+any rewards paid to the sortition pool
+are immediately divided among the currently present operators
+in proportion to their current weight in the pool.
+This requires that joining and leaving the sortition pool quickly is prevented
+so that operators can't pop in for rewards and leave afterwards,
+but requires very little bookkeeping or expensive calculations.
+
+The pool maintains a single global variable for rewards:
+`accumulatedWeightReward`.
+In addition, the total weight of the sortition pool `poolWeight` can be queried.
+Whenever `r` tokens are paid to the pool,
+the accumulator is increased: `accumulatedRewardPerWeight += r / poolWeight`.
+
+Each operator maintains 2 variables for their rewards:
+`weightRewardCollected` and `rewardAvailable`.
+In addition, each operator's `weight` can be queried.
+Whenever an operator's weight changes from `oldWeight` to `newWeight`,
+the rewards earned with the old weight are collected:
+`rewardAvailable += (accumulatedWeightReward - weightRewardCollected) * oldWeight`,
+`weightRewardCollected = accumulatedWeightReward`
+and `weight = newWeight`.
+
+At any time, an operator can withdraw their collected rewards,
+transferring `rewardAvailable` T tokens to them,
+and setting their `rewardAvailable = 0`.
+
+=== Wallet action rewards
+
+Whenever a wallet is required to perform some transaction on-chain,
+a _transaction reward_ is reserved for the wallet.
+When the transaction is correctly submitted,
+the transaction reward is divided
+between the submitter and the rest of the wallet members.
+If the transaction is submitted immediately,
+the full transaction reward is paid out
+and the non-submitter members receive a larger share.
+If the transaction is submitted later,
+the submitter gets a larger reward,
+the other members receive less,
+and some of the reward is withheld from the wallet members altogether.
+The withheld amount is either paid to the sortition pool,
+or kept for future transaction rewards.
+
+For a precise calculation,
+we can use e.g. the Random Beacon entry submission reward rules.
+
+==== Earmarking
+
+Because wallet actions are tied to specific BTC deposits or redemptions,
+it is easy to earmark a specific amount of tBTC for the actions.
+As an example,
+if depositing some amount of BTC takes 0.30% of the deposit as a fee,
+this fee could be divided as follows:
+
+- 0.05% for sweeping the deposit
+- 0.05% for redeeming the deposit (or interwallet transfers)
+- 0.17% for sortition pool members
+- 0.03% for maintenance, e.g. new wallet creation
+
+If someone were to deposit 1 BTC,
+the fee would be 0.003 BTC
+of which 0.0005 BTC would be earmarked for sweeping
+and another 0.0005 BTC for redemption.
+These fees could be collected in the form of TBTC tokens
+immediately upon sweeping the deposit transaction
+and either paid out (sweeping reward)
+or kept in the wallet for future use (redemption reward).
+The remaining 0.002 TBTC
+would be used for sortition pool and maintenance rewards.
```

Comment:
Do we want these allocations to be governable?

Reply:
Yes, I think. I can add a more thorough specification for the fees.

Reply:
I think we should mark it explicitly in the RFC they are expected to be governable so that we do not lose it during the implementation.